jina/types/score/__init__.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+
+from ...excepts import BadNamedScoreType
+from ...proto import jina_pb2
+
+__all__ = ['NamedScore']
+
+
+class NamedScore:
+
+    def __init__(self, score: Optional[jina_pb2.NamedScoreProto] = None,
+                 copy: bool = False, **kwargs):
```

Comment:
One of the points having `NamedScore` is to have a more user-friendly interface, where it is directly obvious, how to use it. When we use reflection instead of explicit getter/setter or properties, this value is lost.

I am fine with using reflection deep in Jina. Anyhow, at the type level which is really close to the user, I'd love to be as explicit as possible.